### PR TITLE
mds: set rctime on new system inode

### PR DESCRIPTION
**Pull request by batrick**
Signed-off-by: Patrick Donnelly <pdonnell@redhat.com>

### CODE REVIEW

**Comment by batrick** [comment]
Has a bug:

````
2018-09-30 21:01:22.679 7fe78b544700 10 MDSIOContextBase::complete: 13C_IO_OFT_Save
2018-09-30 21:01:22.679 7fe78ed4b700 -1 /home/jenkins-build/build/workspace/ceph-dev-new-build/ARCH/x86_64/AVAILABLE_ARCH/x86_64/AVAILABLE_DIST/centos7/DIST/centos7/MACHINE_SIZE/huge/release/14.0.0-3786-gf334317/rpm/el7/BUILD/ceph-14.0.0-3786-gf334317/src/mds/CDir.cc: In function 'bool CDir::check_rstats(bool)' thread 7fe78ed4b700 time 2018-09-30 21:01:22.678527
/home/jenkins-build/build/workspace/ceph-dev-new-build/ARCH/x86_64/AVAILABLE_ARCH/x86_64/AVAILABLE_DIST/centos7/DIST/centos7/MACHINE_SIZE/huge/release/14.0.0-3786-gf334317/rpm/el7/BUILD/ceph-14.0.0-3786-gf334317/src/mds/CDir.cc: 296: FAILED ceph_assert(nest_info.rsubdirs == fnode.rstat.rsubdirs)

 ceph version 14.0.0-3786-gf334317 (f334317efd7226aa12c486320c258c8dcde469cd) nautilus (dev)
 1: (ceph::__ceph_assert_fail(char const*, char const*, int, char const*)+0x153) [0x7fe79b4ed979]
 2: (ceph::__ceph_assertf_fail(char const*, char const*, int, char const*, char const*, ...)+0) [0x7fe79b4edb40]
 3: (CDir::check_rstats(bool)+0x11de) [0x55e2659e786e]
 4: (CInode::finish_scatter_gather_update(int)+0xe3a) [0x55e265a2144a]
 5: (Locker::scatter_writebehind(ScatterLock*)+0x124) [0x55e26597a164]
 6: (Locker::simple_sync(SimpleLock*, bool*)+0x32a) [0x55e26597a97a]
 7: (Locker::scatter_nudge(ScatterLock*, MDSInternalContextBase*, bool)+0x4bd) [0x55e26597f01d]
 8: (Locker::scatter_tick()+0x1f3) [0x55e26597f873]
 9: (Locker::tick()+0x9) [0x55e265995f19]
 10: (MDSRankDispatcher::tick()+0x209) [0x55e26580a349]
 11: (FunctionContext::finish(int)+0x2c) [0x55e2657f4d8c]
 12: (Context::complete(int)+0x9) [0x55e2657f2c59]
 13: (SafeTimer::timer_thread()+0x189) [0x7fe79b58ec39]
 14: (SafeTimerThread::entry()+0xd) [0x7fe79b59101d]
 15: (()+0x7e25) [0x7fe799178e25]
 16: (clone()+0x6d) [0x7fe798041bad]

2018-09-30 21:01:22.682 7fe78ed4b700 -1 *** Caught signal (Aborted) **
````

/ceph/teuthology-archive/pdonnell-2018-09-30_20:23:00-fs-wip-pdonnell-testing-20180930.174832-distro-basic-smithi/3088348/remote/smithi121/log/ceph-mds.a.log.gz

File: src/mds/MDCache.cc [hunk 1]
```diff
@@ -416,13 +417,12 @@ void MDCache::create_empty_hierarchy(MDSGather *gather)
   adjust_subtree_auth(rootdir, mds->get_nodeid());   
   rootdir->dir_rep = CDir::REP_ALL;   //NONE;
 
-  rootdir->fnode.accounted_fragstat = rootdir->fnode.fragstat;
-  rootdir->fnode.accounted_rstat = rootdir->fnode.rstat;
-
-  root->inode.dirstat = rootdir->fnode.fragstat;
-  root->inode.rstat = rootdir->fnode.rstat;
-  ++root->inode.rstat.rsubdirs;
```
Comment (ukernel) [inline src/mds/MDCache.cc]: I think the bug is caused by removing this line

Comment (batrick) [reply src/mds/MDCache.cc]: I've fixed the issue I found here @ukernel . See the comment in the code.

Comment (ukernel) [reply src/mds/MDCache.cc]: I don't see any difference in wip-pdonnell-testing-20181005.225845

Comment (batrick) [reply src/mds/MDCache.cc]: I'm saying the commits you reviewed already had been amended to fix the issue. I haven't see the failure in testing since.

**Comment by smithfarm** [comment]
@batrick Is the first "small refactor" commit needed in backports?

**Comment by batrick** [comment]
@smithfarm no it's not

**Pull request merged**